src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -62,15 +62,22 @@ private ITypeSymbol GetSymbolType(ISymbol symbol)
             return symbol.GetSymbolType();
         }
 
-        protected override CompletionItem CreateItem(
+        protected override CompletionItem CreateItem(CompletionContext completionContext,
             string displayText, string displayTextSuffix, string insertionText,
             List<ISymbol> symbols, SyntaxContext context, bool preselect, SupportedPlatformData supportedPlatformData)
         {
             var rules = GetCompletionItemRules(symbols, context, preselect);
             var matchPriority = preselect ? ComputeSymbolMatchPriority(symbols[0]) : MatchPriority.Default;
             rules = rules.WithMatchPriority(matchPriority);
 
-            if (context.IsRightSideOfNumericType)
+            if (!preselect &&
+                completionContext.Trigger.Kind == CompletionTriggerKind.Insertion &&
+                context.IsInArgumentList &&
+                (completionContext.Trigger.Character == ' ' || completionContext.Trigger.Character == '(' || completionContext.Trigger.Character == '['))
```

Comment:
extract and doc.

Reply:
what if we *are* preselecting, but we're still after the `(` trigger character?  something about this boolean check looks wonky.

Reply:
I've named it `ShouldSoftSelectInArgumentList`, which I think conveys what's happening and why the fallthrough is okay. Let me know if you disagree.